test/should_pass/rigid_type_variables_pass.erl

```diff
@@ -0,0 +1,10 @@
+-module(rigid_type_variables_pass).
+
+-compile([export_all, nowarn_export_all]).
+
+-spec id(A) -> A.
+id(X) -> X.
+
+%% Caused a crash in int_type_to_range/1 on {var, _, 'A'}.
+%% It's now fixed by assuming that a type var means any integer whatsoever.
+foo() -> -id(42).
```

Comment:
Well, the type variable can also be `any()` or `float()`. What we assume is that it's a type that if an integer is included, its negation is also included, or sth like that.... We should perhaps add some constraints on the negated variable, but I don't know how that would look like or how it could work.